service/raspbian-install-device-agent.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root"
+    exit 1
+fi
+
+# Node.js version
+MIN_NODEJS=16
+
+# Update package list and upgrade installed packages
+apt-get update
+
+# Helper functions to test for existence of npm
+function HAS_NPM {
+    if [ -x "$(command -v npm)" ]; then return 0; else return 1; fi
+}
+# Install Node.js and npm if they do not exist
+if [ -x "$(command -v node)" ]; then
+    echo "NodeJS found"
+    VERSION=$(node --version | cut -d "." -f1 | cut -d "v" -f2)
+
+    if [[ $VERSION -ge $MIN_NODEJS ]]; then
+        echo "**************************************************************"
+        echo " NodeJS Version $MIN_NODEJS or newer found "
+        echo "**************************************************************"
+    else
+        echo "**************************************************************"
+        echo " You need NodeJS $MIN_NODEJS or newer, please upgrade "
+        echo "**************************************************************"
+        exit 1
+    fi
+else
+    echo "**************************************************************"
+    echo " No NodeJS found"
+    echo " Do you want to install NodeJS $MIN_NODEJS?"
```

Comment:
This should probably install at least NodeJS 18 if installing from new

Reply:
```suggestion
    echo " Do you want to install NodeJS 18?"
```